app/assets/javascripts/controllers/tree_view_controller.js

```diff
@@ -16,8 +16,41 @@
         });
         vm.$scope.$apply();
       }
+
+      if (payload.updateTrees && _.isObject(payload.updateTrees) && Object.keys(payload.updateTrees).length > 0) {
+        vm.updateTrees(payload.updateTrees);
+        vm.$scope.$apply();
+      }
     });
 
+    vm.updateTrees = function(data) {
```

Comment:
Let's name it `updateNode` or `updateTreeNode` (or plural) as it makes more sense.